### PR TITLE
[C3] Use workers template for `create-hono`

### LINKED ISSUE #3477: 🚀 Feature Request: C3 - use `cloudflare-workers` template for Hono
## tl;dr

What about use `--template cloudflare-workers` option for a Hono command.

---

Hi, this is about C3.

C3 is a great tool to kick-start a project for Workers/Pages. But as the creator of Hono, I've noticed a point of confusion. When we choose "Hono" under the type "Website or web app", C3 runs the `create hono` command with the `--template cloudflare-pages` option:

```
npx create-hono@latest my-app --template cloudflare-pages
```

In reality, the Pages template isn't the best use cases for Hono. It comes with a `functions` endpoint built with Hono and only includes an `index.html` file that links to the endpoint. This might be a good skeleton for developers to build a React application from scratch, but it isn't universally applicable.

My suggestion is to use the `cloudflare-workers` template:

```
npx create-hono@latest my-app --template cloudflare-workers
```

This is a simple change:

https://github.com/cloudflare/workers-sdk/blob/d96cc858d9f90db75a54efc438b17df763be2cc9/packages/create-cloudflare/src/frameworks/hono/index.ts#L13

The `cloudflare-workers` template is minimal but serves as a great starting point for building web APIs. This covers Hono's ideal use case.

However, this template isn't for Pages, it's for Workers. So, it might not fit perfectly under the "Website or web app" category if that category is primarily for Pages. Perhaps we should create a "Web APIs / Backend" category, although the number of templates that would fit into this new category might be limited (itty-router might be another candidate). Alternatively, we could keep it under "Website or web app" and just include Hono there. This is mainly a wording issue, as from another perspective, an app built with Hono is indeed a "web app".

---

But it might not only be a matter for Hono. We could consider creating a "Web API / Backend" category.

### PR DESCRIPTION
**Pull request by yusukebe**
Fixes #3477

**What this PR solves / how to test:**

Currently, in C3, when you choose "Hono" from "Website or web app", it creates the project using the `create-hono` command with the `--template cloudflare-pages` option:

```
npx create-hono@latest my-app --template cloudflare-pages
```

But as I mentioned in #3477, using the template for *Workers* is more appropriate for Hono. In this PR, I've switched to using the `--template cloudflare-workers` option for the `create-hono` command and implemented the following related changes:

* Added an optional `type?` property to `FrameworkConfig`, which can be either `"pages"` or `"workers"`.
* Set the `type` as "`workers`" for Hono.
* If the `type` is set to `workers`, the project isn't created, and a link to `https://developers.cloudflare.com/workers` is displayed instead.

I initially considered whether we should create a separate category for "Web APIs / Backend", but I believe the solution proposed in this PR is better.

**Need to consider**

That the "workers framework" is under "pages" is causing some confusion. It might be worth to refactor by such renaming of some elements. But, I don't have any good ideas at the moment. I'd like to know your opinions.

**What is my motivation?**

I'm planning to write a tutorial for the official Cloudflare documentation. I want to use C3 in this tutorial, but there's currently no suitable way to create a Workers skeleton using Hono. The current Hono template is designed for Pages, which isn't appropriate for this use case.

**Author has included the following, where applicable:**

- [ ] Tests
- [x] Changeset ([Changeset guidelines](https://github.com/cloudflare/workers-sdk/blob/main/CONTRIBUTING.md#changesets))

**Reviewer is to perform the following, as applicable:**

- Checked for inclusion of relevant tests
- Checked for inclusion of a relevant changeset
- Checked for creation of associated docs updates
- Manually pulled down the changes and spot-tested

### CODE REVIEW

**Comment by yusukebe** [comment]
Or we can add the "Hono Worker" coice in the top:

Screenshot 2023-07-03 at 18 13 05

**Review by IgorMinar — COMMENTED** [review]
hey @yusukebe, can we please discuss this with @nevikashah and @admah when they are back at work? Let's not merge this until then. There is a lot of internal discussion around the menu structure and template types, so let's add this PR as something we should discuss as a team.

**Comment by yusukebe** [comment]
🙆‍♂️ thanks @IgorMinar

**Review by admah — APPROVED** [review]
Thanks for the PR @yusukebe. This looks like a good solution to the issue for now. We will separately consider menu structure and the best way to surface frameworks that more closely align with Workers.

**Comment by yusukebe** [comment]
Hi @admah !

Thank you for reviewing!

Is everything approved, and should I proceed to merge this myself?

**Comment by admah** [comment]
Can I get an additional review from @jculvey or @RamIdeas? Thanks!

**Review by jculvey — APPROVED** [review]

**Pull request merged**
